include/aos/iam/certmodule.hpp

```diff
@@ -0,0 +1,380 @@
+ */
+constexpr auto cInvalidKeysPerModule = AOS_CONFIG_CERTHANDLER_INVALID_KEYS_PER_MODULE;
+
+/**
+ * Max number of IAM certificates per module.
+ */
+constexpr auto cCertsPerModule = AOS_CONFIG_CERTHANDLER_MAX_CERTS_PER_MODULE;
+
+/**
+ * Public/private keys generating algorithm
+ */
+class KeyGenAlgorithmType {
+public:
+    enum class Enum { eRSA };
+
+    static const Array<const char* const> GetStrings()
+    {
+        static const char* const sContentTypeStrings[] = {"RSA"};
+        return Array<const char* const>(sContentTypeStrings, ArraySize(sContentTypeStrings));
+    };
+};
+
+using KeyGenAlgorithmEnum = KeyGenAlgorithmType::Enum;
+using KeyGenAlgorithm = EnumStringer<KeyGenAlgorithmType>;
+
+/**
+ * Extended key usage type.
+ */
+class ExtendedKeyUsageType {
+public:
+    enum class Enum { eClientAuth, eServerAuth };
+
+    static const Array<const char* const> GetStrings()
+    {
+        static const char* const sContentTypeStrings[] = {"clientAuth", "serverAuth"};
+        return Array<const char* const>(sContentTypeStrings, ArraySize(sContentTypeStrings));
+    };
+};
+
+using ExtendedKeyUsageEnum = ExtendedKeyUsageType::Enum;
+using ExtendedKeyUsage = EnumStringer<ExtendedKeyUsageType>;
+
+/**
+ * ModuleConfig module configuration.
+ */
+struct ModuleConfig {
+    /**
+     * Key generating algorithm.
+     */
+    KeyGenAlgorithm mKeyGenAlgorithm;
+    /**
+     * Maximum number of certificates for module.
+     */
+    unsigned mMaxCertificates = 0U;
+    /**
+     * Extra extensions needed for CSR. Current supported values: [clientAuth, serverAuth]
+     */
+    StaticArray<ExtendedKeyUsage, cModuleKeyUsagesMaxCount> mExtendedKeyUsage;
+    /**
+     * Alternative DNS names.
+     */
+    StaticArray<StaticString<crypto::cDnsNameLen>, crypto::cAltDnsNamesCount> mAlternativeNames;
+    /**
+     * Skip certificate chain validation.
+     */
+    bool mSkipValidation = false;
+};
+
+/**
+ * General certificate information.
+ */
+struct CertInfo {
+    /**
+     * Certificate issuer name.
+     */
+    StaticString<crypto::cCertificateIssuerLen> mIssuer;
+    /**
+     * Certificate serial number.
+     */
+    StaticString<crypto::cCertificateSerialNumberLen> mSerial;
+    /**
+     * Certificate url.
+     */
+    StaticString<cURLLen> mCertURL;
+    /**
+     * Certificate's private key url.
+     */
+    StaticString<cURLLen> mKeyURL;
+    /**
+     * Certificate expiration time.
+     */
+    time::Time mNotAfter;
+    /**
+     * Compares certificate info.
+     *
+     * @param certInfo info to compare.
+     * @return bool.
+     */
+    bool operator==(const CertInfo& certInfo) const;
+    /**
+     * Compares certificate info.
+     *
+     * @param certInfo info to compare.
+     * @return bool.
+     */
+    bool operator!=(const CertInfo& certInfo) const { return !operator==(certInfo); }
+};
+
+/**
+ * Platform dependent secure certificate storage.
+ */
+class SecureStorageItf {
```

Comment:
Probably better to name it as `HSMItf`. HSM is a commonly used acronym: https://en.wikipedia.org/wiki/Hardware_security_module